pkg/config/lima_config_applier.go

```diff
@@ -107,9 +111,65 @@ func NewLimaApplier(
 		limaDefaultConfigPath:  limaDefaultConfigPath,
 		limaOverrideConfigPath: limaOverrideConfigPath,
 		systemDeps:             systemDeps,
+		finchConfigPath:        finchConfigPath,
 	}
 }
 
+func loadFinchConfig(fs afero.Fs, finchConfigPath string, logger flog.Logger, systemDeps LoadSystemDeps, mem fmemory.Memory) (*Finch, error) {
+	b, err := afero.ReadFile(fs, finchConfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read finch.yaml: %w", err)
+	}
+
+	var cfg Finch
+	if err := yaml.Unmarshal(b, &cfg); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal config file: %w", err)
+	}
+	if err := validate(&cfg, logger, systemDeps, mem); err != nil {
+		return nil, fmt.Errorf("failed to validate config file: %w", err)
+	}
+
+	return &cfg, nil
+}
+
+func (lca *limaConfigApplier) ModifyFinchConfig(fs afero.Fs, logger flog.Logger, cpus int, memory string) (bool, error) {
+	var isConfigUpdated bool
+
+	systemDeps := system.NewStdLib()
+	mem := fmemory.NewMemory()
+
+	finchCfg, err := loadFinchConfig(fs, lca.finchConfigPath, logger, systemDeps, mem)
+	if err != nil {
+		return isConfigUpdated, fmt.Errorf("failed to load config: %w", err)
+	}
+
+	if cpus == 0 && memory == "" {
```

Comment:
suggest having separate checks, may be in a different function and switch cases may be appropriate as in the future we can have more options.

Reply:
Ok, I will try to handle input validation in a separate function.

Reply:
Fixed logic for validation.